packages/react-charting/src/components/TreeChart/TreeChart.base.tsx

```diff
@@ -0,0 +1,293 @@
+      .attr('x', () => {
+        return xCoordinate + rectangleWidth / 3.4;
+      })
+      .text(() => {
+        return subname;
+      });
+  }
+  // Creates a rectangular path from parent to the child nodes
+  public addLinktoNodes(child: any, parent: any, leaf: boolean, rectWidth: number, gap: number): any {
+    let path = `M${child.x + rectWidth / 2},${child.y - gap} H${parent.x + rectWidth / 2} V${parent.y}`;
+    let leafpath = `M${parent.x + rectWidth / 2},${parent.y} V${parent.y + gap * 5} H${parent.x - gap / 2} H${
+      parent.x + rectWidth + gap / 2
+    }`;
+    return leaf ? leafpath : path;
+  }
+}
+// Create child class to Add Tree component based on treeHeight
+class LayeredTree extends StandardTree {
+  public composition: number | undefined;
+  constructor(
+    treeData: IDataStructure,
+    composition: number | undefined,
+    svg: Selection<SVGSVGElement | null, unknown, null, undefined>,
+  ) {
+    super(treeData, svg);
+    this.composition = composition;
+  }
+  public createTree() {
+    let root = hierarchy(this.treeData, function (d: any) {
+      return d.children;
+    });
+
+    let treeHeight = root?.height + 1;
+
+    // Create tree layout, for 2 layered width is 150, 3 layered width is 75
+    let treemap = tree().nodeSize([treeHeight === 2 ? 150 : 60, 50]);
+
+    // Assigns the x and y position for the nodes
+    let treeData = treemap(root);
+
+    // Compute the new tree layout.
+    let nodes = treeData.descendants();
+
+    // Normalize for fixed-depth and width
+    nodes.forEach(function (d) {
+      d.y = d.depth * 120 + 6;
+      d.x += 400;
+    });
+
+    // <------------------ Nodes section ------------------>
+
+    // Create tree data structure
+    const BFS: Array<ITreeDataStructure> = [];
+    let TreeID: number = 0;
+    root.eachBefore((d: any) => {
+      // make id to find parentID from parent object
+      d['id'] = TreeID;
+      BFS.push({
+        id: `${TreeID}`,
+        children: d.children,
+        visualID: `depth: ${d.depth}`,
+        dataName: `${d.data.name}`,
+        subName: `${d.data.subname}`,
+        fill: d.data.fill,
+        x: d.x,
+        y: d.y,
+        parentID: d.parent?.id,
+      });
+      TreeID++;
+    });
+
+    let rectHeight = 60,
+      rectWidth = 120,
+      gap: number = 20;
+    let parentSet = new Set();
+
+    for (const d of BFS) {
+      // check for leaf nodes
+      if (!d.children && !parentSet.has(d.parentID) && treeHeight === 3) {
+        const newWidth = 70,
+          newHeight = 60;
+        parentSet.add(d.parentID);
+
+        let children: any = BFS[d.parentID]?.children;
+        // if the parent has 1 child
+        if (children.length === 1) {
+          this.addNodeShapetoSVG(d.dataName, d.subName, d.x, d.y, d.fill, rectWidth, newHeight);
+        }
+
+        // if the parent has more than 2 child
+
+        let dx1: number = BFS[d.parentID]?.x - newWidth * 1.6,
+          dy: number = children[0]?.y,
```

Comment:
move this code inside if?

Reply:
fixed (added declaration inside if condition for more than 2 child)